### PR TITLE
Event bus shared

### PR DESCRIPTION
**Pull request by hendrikebbers**
Changing event bus API based on the following points:

- Shared event bus support for different systems
- event context added
- filter for event context

Currently missing:
- [x] doc for spec
- [x] test in distributed environment
- [x] test JSON parser
- [x] sample

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/709)

### CODE REVIEW

**Comment by hendrikebbers** [comment]
Implementation for https://github.com/canoo/dolphin-platform/issues/636

File: platform-examples/distribution-sample/distribution-server-hazelcast/src/main/java/com/canoo/platform/samples/distribution/server/hazelcast/HazelcastServer.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    private String hazelcastGroup;
+
+    @Value("${dolphinPlatform.hazelcast.server.port}")
+    private int hazelcastPort;
+
+    private HazelcastInstance hazelcastInstance;
+
+    @PostConstruct
+    public void postConstruct() {
+        Config config = new Config();
```
Comment (hendrikebbers) [inline platform-examples/distribution-sample/distribution-server-hazelcast/src/main/java/com/canoo/platform/samples/distribution/server/hazelcast/HazelcastServer.java]: add final keyword

File: platform-examples/distribution-sample/distribution-server-spring-instance-two/src/main/java/com/canoo/platform/samples/distribution/server/DistributionServerInstanceTwo.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.canoo.platform.remoting.server.spring.DolphinPlatformRemotingApplication;
+import com.canoo.platform.samples.distribution.server.base.DistributionServerConfiguration;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.annotation.Import;
+
+@DolphinPlatformRemotingApplication
+@Import(DistributionServerConfiguration.class)
+public class DistributionServerInstanceTwo {
+
+    public static void main(String... args) {
```
Comment (hendrikebbers) [inline platform-examples/distribution-sample/distribution-server-spring-instance-two/src/main/java/com/canoo/platform/samples/distribution/server/DistributionServerInstanceTwo.java]: use final keyword

File: documentation/src/docs/asciidoc/event-bus.adoc [hunk 3]
```diff
@@ -254,4 +254,53 @@ NOTE: Currently this values can only be specified in the `dolphin.properties` fi
...
 possible for the Hazelcast integration (see https://github.com/canoo/dolphin-platform/issues/377[this issue]). In
 addtion a already defined Hazelcast client that is provided by Spring or JavaEE as a managed bean can not be used
-(see https://github.com/canoo/dolphin-platform/issues/378[this issue]).
\ No newline at end of file
+(see https://github.com/canoo/dolphin-platform/issues/378[this issue]).
+
+
+=== Specification of the distributed event bus protocol
+
+The distributed event bus communications over hazelcast by sending and receiving events in JSON. To provide
```
Comment (janakmulani) [inline documentation/src/docs/asciidoc/event-bus.adoc]: communicates

Comment (hendrikebbers) [reply documentation/src/docs/asciidoc/event-bus.adoc]: done

**Pull request merged**